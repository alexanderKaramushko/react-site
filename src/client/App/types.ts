import { AuthenticationState } from '@store/reducers/authentication/types';

export type OwnProps = {}

export type StateProps = Pick<AuthenticationState, 'authenticationError' | 'authenticationLoading' | 'decodedJWTToken'>

export type DispatchProps = {};

export type Props = OwnProps & StateProps & DispatchProps;
