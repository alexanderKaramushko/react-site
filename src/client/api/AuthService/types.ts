export interface AuthRequestPayload {
  password: string;
  username: string;
}

export interface AuthResponse {
  access_token: string;
}
