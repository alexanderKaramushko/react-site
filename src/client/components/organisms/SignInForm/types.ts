import { AuthenticationAsyncRequest } from '@store/reducers/authentication/actions.types';
import { AuthenticationState } from '@store/reducers/authentication/types';

export type OwnProps = {}

export type StateProps = Pick<AuthenticationState, 'authenticationError' | 'authenticationLoading'>

export interface DispatchProps {
   authenticate: AuthenticationAsyncRequest;
}

export type Props = OwnProps & StateProps & DispatchProps;
