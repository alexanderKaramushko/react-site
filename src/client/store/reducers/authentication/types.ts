export enum AUTHENTICATION_TYPES {
  AUTHENTICATION_ASYNC_REQUEST = 'AUTHENTICATION_ASYNC_REQUEST',
  AUTHENTICATION_ASYNC_SUCCESS = 'AUTHENTICATION_ASYNC_SUCCESS',
  AUTHENTICATION_ASYNC_FAILURE = 'AUTHENTICATION_ASYNC_FAILURE',
}

export type AuthenticationState = {
  authenticationError: string | null;
  authenticationLoading: boolean;
};
