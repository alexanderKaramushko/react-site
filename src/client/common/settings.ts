export type ThemeType = 'light' | 'dark';

export enum Themes {
    LIGHT = 'light',
    DARK = 'dark',
}
