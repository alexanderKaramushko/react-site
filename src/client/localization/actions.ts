import { setLocale } from 'react-redux-i18n';
import { Dispatch } from 'redux';
import { supportedLocales, fallbackLocale } from './config';

export function setLocaleWithFallback(dispatch: Dispatch) {
    return (desiredLocale: string): void => {
        const finalLocale = Object.keys(supportedLocales).includes(desiredLocale)
            ? desiredLocale
            : fallbackLocale;

        setLocale(finalLocale)(dispatch);
    };
}
