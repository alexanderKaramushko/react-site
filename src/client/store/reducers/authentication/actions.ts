import { createAsyncAction } from 'typesafe-actions';

import { AuthRequestPayload } from '../../../api/AuthService/types';
import { AUTHENTICATION_TYPES } from './types';
import { DecodedJWTToken } from '../../../models/base/DecodedJWTToken';

export const authenticateAsync = createAsyncAction(
  AUTHENTICATION_TYPES.AUTHENTICATION_ASYNC_REQUEST,
  AUTHENTICATION_TYPES.AUTHENTICATION_ASYNC_SUCCESS,
  AUTHENTICATION_TYPES.AUTHENTICATION_ASYNC_FAILURE,
)<AuthRequestPayload, DecodedJWTToken, string>();
