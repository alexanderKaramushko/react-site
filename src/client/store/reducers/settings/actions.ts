import { createAction } from 'typesafe-actions';
import { Themes } from '../../../common/settings';

export const toggleThemeAction = createAction('TOGGLE_THEME', (themeName: Themes) => themeName)<Themes>();
export const toggleTextSizeAction = createAction('TOGGLE_TEXT_SIZE', (size: number) => size)<number>();
