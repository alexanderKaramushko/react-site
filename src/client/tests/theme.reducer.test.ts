import { getType } from 'typesafe-actions';
import { settingsReducer } from '../store/reducers/settings/settings';
import { toggleThemeAction } from '../store/reducers/settings/actions';
import { Themes } from '../store/reducers/settings/types';

describe('With snapshots ', () => {
  it('+++ reducer with shapshot', () => {
    const action = {
      payload: Themes.DARK,
      type: getType(toggleThemeAction),
    };

    expect(settingsReducer(undefined, action)).toMatchSnapshot();
  });

  it('+++ reducer for TOGGLE_THEME', () => {
    const state = { textSize: 100, themeName: Themes.DARK };
    const themeReducer = settingsReducer(state, {
      payload: Themes.DARK,
      type: getType(toggleThemeAction),
    });

    expect(themeReducer).toEqual(state);
  });
});
