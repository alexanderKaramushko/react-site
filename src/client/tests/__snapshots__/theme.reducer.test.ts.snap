// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With snapshots  +++ reducer with shapshot 1`] = `
Object {
  "themeName": "dark",
}
`;
